res/controllers/common-controller-scripts.js

```diff
@@ -442,6 +442,48 @@ bpm.tapButton = function(deck) {
 //     print("Script: BPM="+average+" setting to "+fRateScale);
 }
 
+// color - Basic Object used for convient interface with the hotcue_color ControlObjects
+color = {
+  // member should be set directly but only in the [0;255] range.
+  red: 0,
+  green: 0,
+  blue: 0,
+  alpha: 255,
+}
+
+
+/* -------- ------------------------------------------------------
+      color.fromCO
+   Purpose: Setter that splits the combined colors into their components
+            and sets the color components in the object
+   Input:   combined color value
+   Output:  -
+   -------- ------------------------------------------------------ */
+
+color.fromCO = function(color) {
```

Comment:
The controller script interface is still not done. Before the CO represented the RGBA color as an int and this function was useful.

Now the CO represents the color Id, and I need to write a new interface for the JS code.

Sorry for not warning about it.

Reply:
We can offer three additional CO that carry the customized rgb value (one component per CO). Thus, scripters can directly get the values.
We will also provide the predefined color id as CO.

Additionally to this, I will create new functions on the engine so scripts can also get a list of the predefined colors, including its name, default rgba and current custom rgba. This will be useful for mappings that can modify cue colors.

Reply:
Do we relay need both? Id + interface to fetch the default color should be IMHO sufficient.

Reply:
Yes, one CO for the Id it's enough.

Reply:
Yes, I'll will remove this when making the proper script api.

Reply:
I would prefere to remove it because it is highly misleading for other readers.

Reply:
Done